nexus/handle.go

```diff
@@ -0,0 +1,155 @@
+func (h *OperationHandle) GetInfo(ctx context.Context, options GetInfoOptions) (*OperationInfo, error) {
+	url, err := h.client.joinURL(h.Operation, h.ID)
+	if err != nil {
+		return nil, err
+	}
+	httpReq, err := http.NewRequestWithContext(ctx, "GET", url.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	if options.Header != nil {
+		httpReq.Header = options.Header.Clone()
+	}
+
+	httpReq.Header.Set(headerUserAgent, userAgent)
+	response, err := h.client.Options.HTTPCaller(httpReq)
+	if err != nil {
+		return nil, err
+	}
+
+	// Do this once here and make sure it doesn't leak.
+	body, err := readAndReplaceBody(response)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, newUnexpectedResponseError(fmt.Sprintf("unexpected response status: %q", response.Status), response, body)
+	}
+
+	return operationInfoFromResponse(response, body)
+}
+
+// GetResultOptions are Options for [OperationHandle.GetResult].
+type GetResultOptions struct {
+	// Header to attach to the HTTP request. Optional.
+	Header http.Header
+	// Duration to wait for operation completion. Zero or negative value implies no wait.
+	Wait time.Duration
+}
+
+// GetResult gets the result of an operation, issuing a network request to the service handler.
+//
+// By default, GetResult returns (nil, [ErrOperationStillRunning]) immediately after issuing a call if the operation has
+// not yet completed.
+//
+// Callers may set GetResultOptions.Wait to a value greater than 0 to alter this behavior, causing the client to long
+// poll for the result issuing one or more requests until the provided wait period exceeds, in which case (nil,
+// [ErrOperationStillRunning]) is returned.
+//
+// The wait time is capped to the deadline of the provided context.
+//
+// ⚠️ If a response is returned, its body must be read in its entirety and closed to free up the underlying connection.
+func (h *OperationHandle) GetResult(ctx context.Context, options GetResultOptions) (*http.Response, error) {
+	url, err := h.client.joinURL(h.Operation, h.ID, "result")
+	if err != nil {
+		return nil, err
+	}
+	httpReq, err := http.NewRequestWithContext(ctx, "GET", url.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	if options.Header != nil {
+		httpReq.Header = options.Header.Clone()
+	}
+	httpReq.Header.Set(headerUserAgent, userAgent)
+
+	startTime := time.Now()
+	for {
+		var wait time.Duration
+		if options.Wait > 0 {
+			wait = options.Wait - time.Since(startTime)
+			if wait < 0 {
+				return nil, ErrOperationStillRunning
+			}
+		}
+		response, err := h.client.sendGetOperationResultRequest(ctx, httpReq, wait)
+		if err != nil {
+			if errors.Is(err, ErrOperationStillRunning) {
+				if options.Wait > 0 {
+					continue
+				} else {
+					return nil, ErrOperationStillRunning
+				}
+			}
+			return nil, err
+		}
+		return response, nil
+	}
+}
+
+// CancelOptions are options for [OperationHandle.Cancel].
+type CancelOptions struct {
```

Comment:
I wonder if we should prefix these with `Operation` to avoid potential ambiguity for other uses of, say, "cancel options".

Reply:
Hmm.. they match the handle methods, but yes, I could add that to match `StartOperationOptions` and `ExecuteOperationOptions`.

Reply:
Changed all handle options to include the work `Operation`.